### PR TITLE
[risk=low][no ticket] Update deprecated CircleCI docker image to "next-gen"

### PR DESCRIPTION
**Pull request by jmthibault79**
Our current image (and all beginning with `circleci/`) are deprecated: https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ci/Dockerfile.circle_build [hunk 1]
```diff
@@ -13,7 +13,8 @@
 #
 # We use OpenJDK 8, Node, and some common browsers from CircleCI's base image
 # see: https://circleci.com/docs/2.0/circleci-images/#language-image-variants
-FROM circleci/openjdk:8-buster-node-browsers
```
Comment (jmthibault79) [inline ci/Dockerfile.circle_build:16]: Specifying the Debian distribution (e.g. `buster`) is no longer supported.  The new images use Ubuntu LTS.

`-node` is now included in `-browsers` (and this particular image installs Node 16)

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 2]
```diff
@@ -406,6 +406,7 @@ public DbUser createUser(
     if (dbAddress != null) {
       dbAddress.setUser(dbUser);
     }
+
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:409]: I'll revert this after API tests pass

File: ui/src/app/pages/analysis/runtime-panel.tsx [hunk 3]
```diff
@@ -1558,9 +1558,9 @@ const RuntimePanel = fp.flow(
       ],
       [() => true, () => PanelContent.Customize],
     ])([workspace.billingStatus, currentRuntime, status]);
+
```
Comment (jmthibault79) [inline ui/src/app/pages/analysis/runtime-panel.tsx:1561]: I'll revert this after UI tests pass

**Review by calbach — APPROVED** [review]
Nice, thanks!

File: .circleci/config.yml [hunk 4]
```diff
@@ -16,7 +16,7 @@ parameters:
     default: "circle-deploy-account@all-of-us-workbench-test.iam.gserviceaccount.com"
   workbench-image:
     type: string
-    default: "allofustest/workbench:buildimage-0.0.22"
+    default: "allofustest/workbench:buildimage-0.0.23"
   db-image:
     type: string
     default: "circleci/mysql:5.7"
```
Comment (calbach) [inline .circleci/config.yml:22]: Looks like this can also be updated to cimg/mysql:5.7

**Review by aweng98 — APPROVED** [review]

File: .circleci/config.yml (hunk 4 above)
Comment (jmthibault79) [reply .circleci/config.yml]: good catch, done

File: ui/src/app/pages/admin/admin-users.spec.tsx [hunk 5]
```diff
@@ -14,7 +15,11 @@ describe('AdminUsers', () => {
   let props: { profile: Profile; hideSpinner: () => {}; showSpinner: () => {} };
 
   const component = () => {
-    return mount(<AdminUsers {...props} />);
+    return mount(
+      <MemoryRouter>
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-users.spec.tsx:19]: Not clear why the new version of Node is pickier about this, but it's a correct change that we have made for many other tests.

**Pull request merged**